File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/minimum-difference-between-highest-and-lowest-of-k-scores.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        
        nums.sort()
        # 1 4 7 9

        l = 0
        k = k-1
        diff = 100000000000
        for r in range(k, len(nums)):
            diff = min(abs(nums[l] - nums[r]), diff)
            l+=1
        print(diff)
        return diff

File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/minimum-difference-between-highest-and-lowest-of-k-scores.java
class Solution {
    public int minimumDifference(int[] nums, int k) {
        
        Arrays.sort(nums);
        // int i = 0;
        int minDiff = Integer.MAX_VALUE;
        
        for (int i = 0; i <= nums.length - k; i++) {
            int j = i + k - 1;
            int diff = nums[j] - nums[i];
            if (diff < minDiff) {
                minDiff = diff;
            }
        }
        
        return minDiff;
    }
    
}